modules/apps/headless/headless-builder/headless-builder-impl/src/main/resources/com/liferay/headless/builder/internal/batch/headless-builder.batch-engine-data.json

```diff
@@ -284,6 +284,21 @@
 					"type": "oneToMany"
 				}
 			],
+			"objectValidationRules": [
```

Comment:
Hey @matijapetanjek,

I love this approach, but I am not sure it is possible to use it:
1. As far as I know groovy validations are disabled on LXC. Can you double check this with the objects team?
2. The error message cannot be automatically translated to other languages. It would be great if we could just indicate the language key, but I don't think that is supported. Can you check this with the objects team too?

At the moment we are implementing this kind of validations inside the model listener classes in `headless-builder-impl`. 

Also, it would be great if you could write a test for this validation (see the mode listener tests in `headless-builder-test`.

Reply:
Got it, if you're OK with the approach, I'm changing it to model listener.

Reply:
@gabrielwas can you have a look and let us know if those things are possible?

@matijapetanjek for now yeah do the model listener. I think if points 1 and 2 are doable eventually, we will change all in one go.